### DISCUSSION TITLE
Touch command on Sonos

### DISCUSSION THREAD

**Discussion by Kirium** [hklages/node-red-contrib-sonos-plus#137] [Q&A] [answered]
Hi hklages,

I'm using your nodes and thank you. It's working everyday fine !
I would disable the touch command of my sonos during the housekeeping.
Do you have a solution for me ?

Kind regards !

**Comment by hklages** [comment]
Hi Kirium. 

> I'm using your nodes and thank you. It's working everyday fine !

Glad to hear.

What player? Play:1, 3 5 ?  
Do you already use version 5.0.0 of sonos-plus?

**Comment by hklages** [comment, marked as answer]
5.0.2 is out. 
Have a look at universal node: player.get.buttonlockstate and player.set.buttonlockstate. 
It works fine on my Play:3, Play:5 and beam.

--- it seems that SONOS up does not update its status so you can only test it direct on the player. 

Does it work for your?

**Reply by Kirium** [reply]
Thank you so much !

Working very fine on my sonos one's and also on my playbase.

I have a 5.1 (playbase + 2 ones and a sub) and everything is locked. (Except button for enable/disable microphone, but it's normal).

Thank you, the housekeeping woman will no longer be worried working here  :stuck_out_tongue_winking_eye:

**Comment by hklages** [comment]
great! 

And thanks for the coffee!

**Comment by hklages** [comment]
Regarding microphone: with package [sonos-events](https://www.npmjs.com/package/node-red-contrib-sonos-events) you are able to subscribe to changes. If you subscribe to microphone you will be noticed when someone switches it on or off.

**Reply by fricastle** [reply]
Hi I was searching for a way to activate and deactivate the mic with node red. Unfortunately I was not able to find any other way then touch.
My Sonos One is placed very high in the room and I want to activate/deactivate the mic without having to stand on a chair.
Is there a way to do it remotely?
BR

**Comment by hklages** [comment]
It is not possible at the moment because SONOS does not support it. 

I remember an article saying that SONOS will never enable it. They have privacy concerns. Someone remote may open the microphone and listen to conversations. 

So the "push the button" is the one and only choice. You may use a  [switchbot bot](https://www.switch-bot.com/products/switchbot-bot) and link that to your Home automation (such as Home Assistant or even to Node-RED) . That might work - (just an idea - not tested)
